src/draw/nxp/g2d/lv_g2d_buf_map.c

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file lv_g2d_buf_map.c
+ *
+ */
+
+/**
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "lv_g2d_buf_map.h"
+
+#if LV_USE_DRAW_G2D
+#include <stdio.h>
+#include "lv_g2d_utils.h"
+#include "g2d.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static unsigned long _map_hash_function(void * ptr);
+
+static lv_map_item_t * _map_create_item(void * key, struct g2d_buf * value);
+
+static void _map_free_item(lv_map_item_t * item);
+
+static void _handle_collision(unsigned long index, lv_map_item_t * item);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+static lv_buf_map_t * table;
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void g2d_create_buf_map(void)
+{
+    table = (lv_buf_map_t *) lv_malloc(sizeof(lv_buf_map_t));
+    table->size = HASH_TABLE_SIZE;
+    table->count = 0;
+    table->items = (lv_map_item_t **) lv_malloc_zeroed(table->size * sizeof(lv_map_item_t *));
+    table->overflow_list = (lv_array_t **) lv_malloc_zeroed(table->size * sizeof(lv_array_t *));
+
+    for(int i = 0; i < table->size; i++) {
+        table->items[i] = NULL;
+        table->overflow_list[i] = NULL;
+    }
+}
+
+void g2d_free_buf_map(void)
+{
+    for(int i = 0; i < table->size; i++) {
+        lv_array_deinit(table->overflow_list[i]);
```

Comment:
`lv_array_deinit` does not expect `NULL`
```suggestion
        if(table->overflow_list[i]) {
            lv_array_deinit(table->overflow_list[i]);
            lv_free(table->overflow_list[i]);
        }
```

Reply:
Done